legend-pure-code-compiled-core-relational/src/main/resources/core_relational/relational/sqlQueryToString/SQLQueryToString.pure

```diff
@@ -773,19 +785,31 @@ function <<access.private>> meta::relational::functions::sqlQueryToString::proce
    $children->map(child | processJoinTreeNode($child, $tableAlias->cast(@TableAlias), $dbConfig, $format, $extensions))->joinStrings($val, '', '');
 }
 
+
+function <<access.private>> meta::relational::functions::sqlQueryToString::isDefaultSchema(table:Table[1], dbConfig : DbConfig[1]):Boolean[1]
+{
+  ($table.schema.name->length() > 0) && ($table.schema.name == 'default');
+}
+
 function <<access.private>> meta::relational::functions::sqlQueryToString::tableToString(table:Table[1], dbConfig : DbConfig[1]):String[1]
 {
+   let validSchema = if ( $dbConfig.dbType == DatabaseType.Spanner, | isDefaultSchema($table, $dbConfig), | true);
+
+   assert($validSchema, 'Invalid schema specification "' + $table.schema.name + '". Spanner does not support named schemas');
+
    let schemaName = if ($dbConfig.dbType == DatabaseType.Presto,
                         | $table.schema.name->replace('$', '.'),
-                        | $table.schema.name
+                        | if ($dbConfig.dbType == DatabaseType.Spanner,

```

Comment:
With the above check maybe we don't need if block here?

Reply:
Yup. Good catch,